csrc/runtime/executor.cpp

```diff
@@ -208,6 +208,15 @@ void KernelExecutor::compile(
     return;
   }
 
+  const bool has_any_cuda_tensor = std::any_of(
+      fusion->inputs().begin(), fusion->inputs().end(), [](Val* inp) {
+        return inp->isA<TensorView>() && !inp->as<TensorView>()->isCpuScalar();
+      });
+
+  NVF_CHECK(
+      has_any_cuda_tensor,
+      "Expected atleast one input to be on DeviceType::CUDA.");
```

Comment:
```suggestion
      "Expected at least one input to be on DeviceType::CUDA.");
```
Also, since this is an internal error and not a misspecification of the overall Fusion, maybe we should use `NVF_ERROR` instead.